.github/workflows/build-test-release.yml

```diff
@@ -59,7 +59,7 @@ jobs:
       - uses: actions/checkout@v4
       - uses: actions/setup-python@v5
         with:
-          python-version: "3.7"
+          python-version: "3.9"
       - uses: pre-commit/action@v3.0.1
```

Comment:
Can we also update `.pre-commit-config.yaml` with later versions of the tools now that we support Py39 and above?